atom/renderer/api/atom_api_spell_check_client.cc

```diff
@@ -74,14 +96,28 @@ void SpellCheckClient::RequestCheckingOfText(
     const blink::WebString& textToCheck,
     blink::WebTextCheckingCompletion* completionCallback) {
   base::string16 text(textToCheck.Utf16());
+  // Ignore invalid requests.
   if (text.empty() || !HasWordCharacters(text, 0)) {
     completionCallback->DidCancelCheckingText();
     return;
   }
 
-  std::vector<blink::WebTextCheckingResult> results;
-  SpellCheckText(text, false, &results);
-  completionCallback->DidFinishCheckingText(results);
+  // Clean up the previous request before starting a new request.
+  if (pending_request_param_.get()) {
+    pending_request_param_->completion()->DidCancelCheckingText();
+  }
+
+  pending_request_param_.reset(new SpellcheckRequest(text, completionCallback));
+
+  SpellcheckRequest* request = pending_request_param_.release();
```

Comment:
This confuses me. After .release(), future calls to pending_request_param_.get() will return nullptr so I don't see how the conditional on line 106 will ever be true?

As written, pending_request_param_ is acting like a local temporary variable.

_(minor)_ I'd suggest the terser if `(pending_request_param_)` to `if (pending_request_param_.get())` but that doesn't change the more important comment above

Reply:
You are mostly right, this one is confused me also somewhat. So this is basically picking https://cs.chromium.org/chromium/src/components/spellcheck/renderer/spellcheck.cc?sq=package:chromium&dr=C&l=414-423 , I've brought all checks as-is while it behaves as you've described. While I picks up logic, left those checks for precaution just in case - as I am not 100% sure to call those checks are redundant.